### DISCUSSION TITLE
Make greenlet optional

### DISCUSSION THREAD

**Discussion by kasium** [sqlalchemy/sqlalchemy#10194] [Ideas]
As far as I see, greenlet is only used on some platforms for asyncio, right? I use one of these platforms but don't use asyncio, but greenlet is still installed. Would it be possible to make greenlet completely optional, so that it's no longer an install requirement

**Comment by zzzeek** [comment]
I agree, we rejected this at #6136 but now that asyncio SQLAlchemy is well established, it's time (clarification: 2.1 is time.  sometime next year).

**Comment by kasium** [comment]
Cool, thanks a lot!

**Comment by CaselIT** [comment]
This should be on 2.1 thought, otherwise it will break lots of people code

**Comment by zzzeek** [comment]
yes I created #10197 for this w/ 2.1 milestone

**Comment by zzzeek** [comment]
#10197 is done.
